### PR TITLE
TM-3856 Client Profile AIH link

### PR DESCRIPTION
**Pull request by andrews-john**
added agendaitemhistorylink (renamed). first pass at formatting. link works but probably isnt correct

### CODE REVIEW

File: src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const AgendaItemHistoryLink = (props) => (
+  <div className="usa-grid-full section-padded-inner-container">
+    <div className="usa-grid-full">
+      <div className="usa-width-three-fourths agenda-item-history-header">
+        <SectionTitle title="Agenda Item History" icon="user-circle-o" />
+      </div>
+    </div>
+    <div className="usa-grid-full small-link-container agenda-item-history-link">
+      <StaticDevContent>
+        <Link to={`/profile/${props.userRole}/agendaitemhistory/${props.perdet}`}>Link to Agenda Item History</Link>
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx:15]: change text to `Go to Agenda Item History`

File: src/sass/_profile.scss [hunk 2]
```diff
@@ -510,6 +510,19 @@
     font-size: 1.6rem;
     margin-bottom: 5px;
   }
+  .agenda-item-history-link{
```
Comment (SidelineCory24) [inline src/sass/_profile.scss:513]: can add a new line

File: src/sass/_profile.scss [hunk 3]
```diff
@@ -510,6 +510,19 @@
...
+    border-top: 1px solid #AEB0B5;
+    padding-top: 1em;
+    min-width: 100%;
+    display: block;
+    margin-right: 2.35765%;
+    width: 100%;
+    text-align: center;
+  }
+  .agenda-item-history-header{
```
Comment (SidelineCory24) [inline src/sass/_profile.scss:522]: can add a new line

File: src/Components/ProfileDashboard/ProfileDashboard.jsx [hunk 4]
```diff
@@ -153,6 +156,11 @@ const ProfileDashboard = ({
                         <Assignments assignments={userProfile.assignments} />
                       </BoxShadow>
                     }
+                    {(isPublic && showAgendaItemHistory) &&
+                      <BoxShadow className="usa-width-one-whole user-dashboard-section assignments-section">
+                        <AgendaItemHistoryLink perdet={perdet} userRole={userRole} />
+                      </BoxShadow>
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/ProfileDashboard.jsx:162]: I would put this before `showAssignmentHistory` check to match the screenshot

File: src/Containers/ProfilePublic/ProfilePublic.jsx [hunk 5]
```diff
@@ -68,6 +68,7 @@ class ProfilePublic extends Component {
         showAssignmentHistory: false,
         showClassifications: false,
         showSearchAsClient: false,
+        showAgendaItemHistory: false,
```
Comment (SidelineCory24) [inline src/Containers/ProfilePublic/ProfilePublic.jsx:71]: 🤌

File: src/Components/ProfileDashboard/ProfileDashboard.jsx [hunk 6]
```diff
@@ -42,6 +43,8 @@ const ProfileDashboard = ({
           <MediaQueryWrapper breakpoint="screenLgMin" widthType="max">
             {(matches) => {
               const isBidder = () => includes(get(userProfile, 'permission_groups', []), 'bidder');
+              const perdet = get(userProfile, 'perdet_seq_number') || '';
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/ProfileDashboard.jsx:46]: great job adding the fallback!

File: src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+const AgendaItemHistoryLink = (props) => (
+  <div className="usa-grid-full section-padded-inner-container">
+    <div className="usa-grid-full">
+      <div className="usa-width-three-fourths agenda-item-history-header">
+        <SectionTitle title="Agenda Item History" icon="user-circle-o" />
+      </div>
+    </div>
+    <div className="usa-grid-full small-link-container agenda-item-history-link">
+      <StaticDevContent>
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx:14]: was there an ask to add `<StaticDevContent>`?

Comment (andrews-john) [reply src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx:14]: I think I just copied one of the other modules in the public profile, either notifications or bid tracker. they had static dev content in there, so I just snagged it.

Comment (SidelineCory24) [reply src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx:14]: Gotcha. I would look into that `StaticDevContent` component and see how it works and if it's still needed. Always be careful when you do a straight copy/paste. You don't want to bring in something that's not needed.

**Comment by SidelineCory24** [comment]
@andrews-john when you get a chance hit `update branch` so you have the latest dev branch code

File: src/sass/_profile.scss [hunk 8]
```diff
@@ -510,6 +510,21 @@
     font-size: 1.6rem;
     margin-bottom: 5px;
   }
+
+  .agenda-item-history-link{
```
Comment (SidelineCory24) [inline src/sass/_profile.scss:514]: add a space between `link{` -> `link {`

File: src/sass/_profile.scss [hunk 9]
```diff
@@ -510,6 +510,21 @@
     font-size: 1.6rem;
     margin-bottom: 5px;
   }
+
```
Comment (SidelineCory24) [inline src/sass/_profile.scss:513]: double check that all of this styling is being used

File: src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx (hunk 7 above)
Comment (SidelineCory24) [reply src/Components/ProfileDashboard/AgendaItemHistoryLink/AgendaItemHistoryLink.jsx:14]: @andrews-john flagging

**Comment by SidelineCory24** [comment]
Confirming views for each of the permissions
**ao should show**
image

**bureau no show**
image

**cdo should show**
image

**post no show**
image

File: src/Components/ProfileDashboard/ProfileDashboard.jsx [hunk 10]
```diff
@@ -147,6 +150,12 @@ const ProfileDashboard = ({
                         />
                       </BoxShadow>
                     }
+                    {
+                      (isPublic && showAgendaItemHistory) &&
+                      <BoxShadow className="usa-width-one-whole user-dashboard-section assignments-section">
+                        <AgendaItemHistoryLink perdet={perdet} userRole={userRole} />
+                      </BoxShadow>
+                    }
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/ProfileDashboard.jsx:158]: 🥳

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go!🚀

**Pull request merged**